market-service/src/main/java/rs/edu/raf/banka1/services/ListingServiceImpl.java

```diff
@@ -0,0 +1,295 @@
+            JsonNode rootNode = objectMapper.readTree(new URL(apiUrl));
+
+            List<ListingHistoryModel> listingHistoryModels = new ArrayList<>();
+            // Get the "Time Series (Daily)" node
+            JsonNode timeSeriesNode = rootNode.get("Time Series (Daily)");
+            if (timeSeriesNode != null) {
+                Iterator<Map.Entry<String, JsonNode>> fields = timeSeriesNode.fields();
+                while (fields.hasNext()) {
+                    Map.Entry<String, JsonNode> entry = fields.next();
+                    String date = entry.getKey();
+                    JsonNode dataNode = entry.getValue();
+
+                    // Get specific fields from each data node
+                    double open = dataNode.get("1. open").asDouble();
+                    double high = dataNode.get("2. high").asDouble();
+                    double low = dataNode.get("3. low").asDouble();
+                    double close = dataNode.get("4. close").asDouble();
+                    int volume = dataNode.get("5. volume").asInt();
+
+                    // make a new ListingHistoryModel
+                    ListingHistoryModel listingHistoryModel = new ListingHistoryModel();
+                    listingHistoryModel.setTicker(ticker);
+                    listingHistoryModel.setDate(java.time.LocalDate.parse(date, DateTimeFormatter.ofPattern("yyyy-MM-dd")));
+                    listingHistoryModel.setPrice(close);
+                    listingHistoryModel.setAsk(high);
+                    listingHistoryModel.setBid(low);
+                    listingHistoryModel.setChanged(close - open);
+                    listingHistoryModel.setVolume(volume);
+
+
+                    listingHistoryModels.add(listingHistoryModel);
+                }
+            }
+            return listingHistoryModels;
+        }catch (Exception e){
+            e.printStackTrace();
+            return new ArrayList<>();
+        }
+    }
+
+
+    //    if we want to add listing to history for certain day, if we already have saved it, we should just update it
+    @Override
+    public int addListingToHistory(ListingHistoryModel listingHistoryModel) {
+        Optional<ListingHistoryModel> listingHistoryModelOptional = listingHistoryRepository.findByTickerAndDate(listingHistoryModel.getTicker(), listingHistoryModel.getDate());
+        if (listingHistoryModelOptional.isPresent()) {
+            ListingHistoryModel listingHistoryModel1 = listingHistoryModelOptional.get();
+            listingHistoryModel1.setPrice(listingHistoryModel.getPrice());
+            listingHistoryModel1.setAsk(listingHistoryModel.getAsk());
+            listingHistoryModel1.setBid(listingHistoryModel.getBid());
+            listingHistoryModel1.setChanged(listingHistoryModel.getChanged());
+            listingHistoryModel1.setVolume(listingHistoryModel.getVolume());
+
+            listingHistoryRepository.save(listingHistoryModel1);
```

Comment:
zar nije vec napravljen mapper koji ovo radi, a ako nije, zasto nije?

Reply:
Zato što je ovo updatovanje, nije mapiranje iz jedne klase u drugu, ali slažem se da bi moglo lepše, pa sam ga izdvojio u zasebnu funkciju.